shared_model/validators/commands_validator.hpp

```diff
@@ -264,14 +264,14 @@ namespace shared_model {
         void validatePeerAddress(
             ReasonsGroupType &reason,
             const interface::AddPeer::AddressType &address) const {
-          if (address != "localhost") {
-            std::regex ipRegex(
-                "((([0-1]?\\d\\d?)|((2[0-4]\\d)|(25[0-5]))).){3}(([0-1]?\\d\\d?"
-                ")|((2[0-4]"
-                "\\d)|(25[0-5])))");
-            if (not std::regex_match(address, ipRegex)) {
-              reason.second.push_back("Wrongly formed PeerAddress");
-            }
+          // based on https://stackoverflow.com/questions/106179/regular-expression-to-match-dns-hostname-or-ip-address
+          std::regex valid_ipv4(
+              R"((^((([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])):(6553[0-5]|655[0-2]\d|65[0-4]\d\d|6[0-4]\d{3}|[1-5]\d{4}|[1-9]\d{0,3}|0)$))");
+          std::regex valid_hostname(
+              R"((^(.*[a-zA-Z]+.*)(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9]):(6553[0-5]|655[0-2]\d|65[0-4]\d\d|6[0-4]\d{3}|[1-5]\d{4}|[1-9]\d{0,3}|0)$))");
```

Comment:
Looks like ip without explicitly defined port is not valid anymore, is it ok?

Reply:
Yes, IP is mandatory